# Incidents Master Analysis

In [1]:
# Incident Profile Analysis

import pandas as pd

df = pd.read_csv('incidents_master.csv')

print(df.info())

incident_id = df['incident_id']
revenue = df['company_revenue_usd']
attack_vector = df['attack_vector_primary']
downtime_hours = df['downtime_hours']


print(df.describe())
print("------------------------------------")
pd.set_option('display.max_rows', None)

print()

# Number of Attack Vectors
attack_vectors_numbers = attack_vector.value_counts()
print(attack_vectors_numbers)

print()

# Public/Private Companies
company_type = df['is_public_company'].value_counts()
print(company_type)

print()

# Average Downtime Hours
downtime_stat = downtime_hours.agg([
        'count',
        'mean',
        'median',
        'std',
        ])
print(downtime_stat)  
print(downtime_hours.describe())

print()

# Data Compromised Records
data_com_rec = df['data_compromised_records'].describe()
print(data_com_rec)

print()
# Quality and Confidence Tier
confidence_tier = df['confidence_tier'].value_counts()
quality_grade = df['quality_grade'].value_counts()

print(confidence_tier)
print('---------------------')
print(quality_grade)

print()

# Attack Revenue + Downtime Analysis
attack_downtime = df.groupby("attack_vector_primary")["downtime_hours"].count()

print(attack_downtime)


<class 'pandas.DataFrame'>
RangeIndex: 850 entries, 0 to 849
Data columns (total 32 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   incident_id               850 non-null    str    
 1   company_name              850 non-null    str    
 2   company_revenue_usd       850 non-null    float64
 3   country_hq                850 non-null    str    
 4   industry_primary          850 non-null    str    
 5   industry_secondary        153 non-null    str    
 6   employee_count            850 non-null    int64  
 7   is_public_company         850 non-null    bool   
 8   stock_ticker              412 non-null    str    
 9   incident_date             850 non-null    str    
 10  incident_date_estimated   850 non-null    bool   
 11  discovery_date            850 non-null    str    
 12  disclosure_date           850 non-null    str    
 13  attack_vector_primary     850 non-null    str    
 14  attack_vector_seconda

- This program analyzes the datasets per column, analyzing each single data points.

**Analysis**

In [2]:
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 850 entries, 0 to 849
Data columns (total 32 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   incident_id               850 non-null    str    
 1   company_name              850 non-null    str    
 2   company_revenue_usd       850 non-null    float64
 3   country_hq                850 non-null    str    
 4   industry_primary          850 non-null    str    
 5   industry_secondary        153 non-null    str    
 6   employee_count            850 non-null    int64  
 7   is_public_company         850 non-null    bool   
 8   stock_ticker              412 non-null    str    
 9   incident_date             850 non-null    str    
 10  incident_date_estimated   850 non-null    bool   
 11  discovery_date            850 non-null    str    
 12  disclosure_date           850 non-null    str    
 13  attack_vector_primary     850 non-null    str    
 14  attack_vector_seconda

- This contains all the initial information about the incidents masters dataset
- Company details, attack chaines, disclosure dates, and quality grades are included within this dataset

In [3]:
print(df.describe())

       company_revenue_usd  employee_count  data_compromised_records  \
count         8.500000e+02    8.500000e+02              6.020000e+02   
mean          1.031337e+10    5.433418e+04              2.708462e+06   
std           2.106158e+10    1.266599e+05              2.978069e+07   
min           2.424181e+07    6.700000e+01              1.000000e+03   
25%           2.327545e+08    1.087000e+03              1.200500e+04   
50%           1.261375e+09    6.219500e+03              5.613300e+04   
75%           8.692943e+09    4.359475e+04              2.891170e+05   
max           1.488980e+11    1.411332e+06              5.497365e+08   

       downtime_hours  confidence_tier  quality_score  
count      420.000000       850.000000     850.000000  
mean       107.208762         2.215294      79.963047  
std        184.645681         1.202472      12.209540  
min          1.830000         1.000000      50.040000  
25%         25.822500         1.000000      71.727500  
50%         53.

- This describes the descriptive statistics of all numerical data included within the dataset.
- With this function includes:
    - count : the number of valid/non-null data points; assess the completeness of the dataset by showing how many entries are valid.
    - mean : the average
    - std : the standard deviation
    - min : the minimum/lowest value 
    - 25%. 50%, 75% : the quartiles (Q1, Q2 (median), Q3 respectively)
    - max : the maximum/highest value

In [4]:
pd.set_option('display.max_rows', None)

- Only to display all rows possible

In [5]:
# Number of Attack Vectors
attack_vectors_numbers = attack_vector.value_counts()
print(attack_vectors_numbers)

attack_vector_primary
ransomware      228
phishing        147
data_breach     123
apt              84
malware          79
ddos             67
supply_chain     58
trojan           34
backdoor         30
Name: count, dtype: int64


- This contains all numbers of attack vectors within the dataset
- According to the data shown, Ransomware occurred much greater within the 850 incidents, summing up to 228 attacks, with backdoor having the lowest occurrence of 30

In [6]:
company_type = df['is_public_company'].value_counts()
print(company_type)


is_public_company
False    438
True     412
Name: count, dtype: int64


- This data shows that out of 850 companies within the dataset, 412 are within the public sector.

In [12]:
# Average Downtime Hours
print(downtime_hours.describe())
print(downtime_hours.skew())

count     420.000000
mean      107.208762
std       184.645681
min         1.830000
25%        25.822500
50%        53.600000
75%       119.800000
max      1951.620000
Name: downtime_hours, dtype: float64
5.744713468542288


- This data describes the central tendency; mean, median, and standard deviation is shown.
- According to the results, the central tendency is right skewed.

In [13]:
# Data Compromised Records
data_com_rec = df['data_compromised_records'].describe()
print(data_com_rec)


count    6.020000e+02
mean     2.708462e+06
std      2.978069e+07
min      1.000000e+03
25%      1.200500e+04
50%      5.613300e+04
75%      2.891170e+05
max      5.497365e+08
Name: data_compromised_records, dtype: float64


- This result describes the records of data compromised during the incidents. This may include personal details or information that are leaked.
- The mean is also right skewed from the median.